ADR/00xx-related-pipelines.md

```diff
@@ -0,0 +1,104 @@
+
+  * In this case, the bundle image must be rebuilt for the operand image update to be really be evaluated in a functional way.
+  * This is a one-to-many dependency. One operator bundle depends on many operand images.
+* A user team just has **two components that depend on each other in a functional way**. When they propose an update to one component, they want to submit a corresponding change to the second component, and have those tested together before merging both.
+  * In this case, on any given day, component B’s PRs may or may not have dependencies on PRs from component A. Not all changes are linked.
+  * This is a many-to-many dependency. The user may have lots of components that depend on lots of other components.
+
+In theory **any combination of the cases above** could be present in an app. They could have one intermediate base image, many operand images that depend on that, and one or more operator bundles which depend on those.
+
+Today, users work around how complicated it is to manage digests themselves by instead using *floating tags*, which have the benefit of being easy to use - no need to update - but have the problem of being unclear. It's not exactly clear what you're building against if you refer to it by tag. Potentially insecure. We want to make it easy for users to do better.
+
+## Decision
+
+In order to support linking of multiple components, any component dependency will be recorded as part of the Tekton definition to be run for pull request event types.
+
+### User flow
+
+Users will be able to specify dependencies between components by modifying the pull request Tekton definition or by adding configuration to the UI which will result in AppStudio representing the configuration in the Tekton definition. This definition will allow the specification of whether dependent components should be run in parallel or serially when combined.
+
+### Pipelines as Code (PaC)
+
+PaC will expand its design to include orchestration logic. This orchestrator will enable a plugable scheduler which is capable of identifying the dependent repositories and Tekton definitions (composing the metadata required to specify a component in AppStudio). When a component's Tekton definition indicates a dependency on another component, PaC will be able to combine the defined pipelines so that the resulting Pipeline definition builds images in the proper order.
+
+When identifying the pipeline definitions to merge, PaC will first look at the linked components for the same branch where a build was triggered from. If no matching branches are found then the default branch will be used.
+
+PaC will need to recursively expand the DAG to ensure that all dependent components are linked.
+
+### Build service
+
+The build service will be able to detect any component pull specs from earlier in a composite pipeline that differ from a previous Snapshot and replace all references of the previous pull spec with the new pull spec before building the component.
```

Comment:
that differs?

Reply:
I'm not sure how build-service will do this.

The work to be done: "replace some pullspec references with others"

* Should it be the responsibility of tasks in the composite pipeline?
* Should it be the responsibility of PaC when assembling the composite pipeline?

I don't think build-service can intervene at composite-pipeline-assembly time or at composite-pipeline-run-time.

Reply:
I am going to resolve this conversation as the original comment is resolved. I opened a new thread on this to continue the conversation from @ralphbean: https://github.com/redhat-appstudio/book/pull/114/files#r1256363848